tests/resources/test_project.py

```diff
@@ -103,6 +140,108 @@ def test_share_post_content(project, session):
         project.share(project_id=project.uid)
 
 
+def test_publish_resource(project, session):
+    dataset_id = str(uuid.uuid4())
+    dataset = project.datasets.build(dict(
+        id=dataset_id,
+        name="public dataset", summary="test", description="test"
+    ))
+
+    with pytest.raises(NotImplementedError):
+        project.publish(resource=dataset)
+
+
+def test_publish_resource_v3(project_v3, session_v3):
+    dataset_id = str(uuid.uuid4())
+    dataset = project_v3.datasets.build(dict(
+        id=dataset_id,
+        name="public dataset", summary="test", description="test"
+    ))
+    assert project_v3.publish(resource=dataset)
+
+    assert 1 == session_v3.num_calls
+    expected_call = FakeCall(
+        method='POST',
+        path='/projects/{}/published-resources/{}/batch-publish'.format(project_v3.uid, 'DATASET'),
+        json={
+            'ids': [dataset_id]
+        }
+    )
+    assert expected_call == session_v3.last_call
+
+
+def test_pull_in_resource(project, session):
+    dataset_id = str(uuid.uuid4())
+    dataset = project.datasets.build(dict(
+        id=dataset_id,
+        name="public dataset", summary="test", description="test"
+    ))
+
+    with pytest.raises(NotImplementedError):
+        project.pull_in_resource(resource=dataset)
+
+
+def test_pull_in_resource_v3(project_v3, session_v3):
+    dataset_id = str(uuid.uuid4())
+    dataset = project_v3.datasets.build(dict(
+        id=dataset_id,
+        name="public dataset", summary="test", description="test"
+    ))
+    assert project_v3.pull_in_resource(resource=dataset)
+
+    assert 1 == session_v3.num_calls
+    expected_call = FakeCall(
+        method='POST',
+        path='/teams/{}/projects/{}/outside-resources/{}/batch-pull-in'.format(project_v3.team_id, project_v3.uid, 'DATASET'),
```

Comment:
As a string, DATASET should be part of the path.

Reply:
This also looks like a good candidate for an f-string.